### DISCUSSION TITLE
/collection/arraylist.html

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#29] [Announcements]
# /collection/arraylist.html

一份通俗易懂、风趣幽默的Java学习指南，内容涵盖Java基础、Java并发编程、Java虚拟机、Java企业级开发、Java面试等核心知识点。学Java，就认准Java程序员进阶之路

https://tobebetterjavaer.com/collection/arraylist.html

**Comment by stophemo** [comment]
太细了666

**Comment by TravelerEntity** [comment]
#### 关于`fatsRemove()` 方法有一个疑问：
```java
private void fastRemove(int index) {
    modCount++;
    int numMoved = size - index - 1;
    if (numMoved > 0)
    System.arraycopy(elementData, index+1, elementData, index,
                numMoved); //执行完这行代码，要 remove 的那个数据不就已经被后面的数据给覆盖了么
    elementData[--size] = null; // 再执行这一行，为什么不会把新的元素给置空
}
```
![](http://cdn.tobebetterjavaer.com/tobebetterjavaer/images/collection/arraylist-02.png)

**Reply by SeaHorse-1** [reply]
elementData[--size] = null; 是把数组尾部那个置空吧
